addOns/postman/src/main/java/org/zaproxy/addon/postman/ItemDeserializer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.postman;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.zaproxy.addon.postman.Models.Item;
+
+public class ItemDeserializer extends JsonDeserializer<List<Item>> {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public List<Item> deserialize(JsonParser jsonParser, DeserializationContext ctxt)
+            throws IOException {
+        JsonNode itemsNode = jsonParser.getCodec().readTree(jsonParser);
+
+        if (itemsNode.isArray()) {
+            return deserializeArray(itemsNode);
+        } else if (itemsNode.isObject()) {
+            return deserializeObject(itemsNode);
+        }
+
+        return new ArrayList<>();
```

Comment:
Is the node ever a simple string, int, boolean? In those cases is returning an empty arraylist the right choice? [I'm just asking because I know we've hit that elsewhere parsing things like this.]

@thc202 is List.of() any more efficient or correct here?

Reply:
Yes, it would be better to use `List.of()` (if the list is not modified later).

Reply:
Does this mean we shall return an unmodifiable list at the other two places too?

Reply:
No, if the list needs to be modifiable leave the code as-is.

Reply:
It doesn't

Reply:
Then it should use `List.of()` here. I'm not sure which other two places you mean, but it should be simple enough to test/figure out. If they need to be modifiable then use a new ArrayList if not then use List.of()